pkg/cloudprovider/cloudprovider.go

```diff
@@ -319,7 +319,7 @@ func (c *CloudProvider) Name() string {
 }
 
 func (c *CloudProvider) GetSupportedNodeClasses() []status.Object {
-	return []status.Object{&v1alpha2.AKSNodeClass{}}
+	return []status.Object{&v1beta1.AKSNodeClass{}}
```

Comment:
I looked and I think that this API is used just to register a watch on the NodeClass, which means only returning a single (latest version) nodeclass is right here, but just worth confirming with @tallaxes

Reply:
IIRC it is used in core to decide which nodeclasses to act or not act upon. As long as we are confident that (after the change) core won't (and won't need to) even observe v1alpha1 - returning the latest seems fine.

Reply:
Ok, based on my reading I think this is correct so I think we can close this comment.